### DISCUSSION TITLE
Add `PresenceEntity` for the `device_tracker` platform

### DISCUSSION THREAD

**Discussion by jbouwh** [home-assistant/architecture#831] [Entity Models]
# Context
The `device_tracker` already has the `ScannerEntity` and the `TrackerEntity`, both used to track persons.
To track room presence this is not on option.
A current implementations like `mqtt_room` implement a `sensor` entity with `extra_state_attributes` which is not scalable. Other implementation use the `binary_sensor` platform.

# Proposal
To track the use of rooms it would be great to have a `PresenceEntity` the state indicates `presence` if there is someone presence in the room and `no_presence` if the room is empty.
This seems a better alternative than the current `sensors` and `binary_sensors` that implement room presence.

Suggested attributes are:
- `source_type` (required)
- `location_name` (optional)
- `latitude` and `longitude` (optional)
- `battery_level` (optional)

# Aim
The `PresenceEntity` allows integrations to link the person's location to a room.

**Comment by MartinHjelmare** [comment]
Why would latitude and longitude be interesting?

Battery level should not be added. That's a separate sensor measurement.

Why is this better than a binary sensor?

**Reply by jbouwh** [reply]
This would help linking a user to a room.
A room presence sensor could detect a users device. The the linked device tracker of the user could inherit the location of the room presence sensor.

**Reply by jbouwh** [reply]
The use of latitude and longitude is may be less interesting.

**Comment by scop** [comment]
`home`/`not_home` does not seem quite right to me for a presence sensor. `present`/`absent` or `present`/`not_present` would sound better.

**Reply by jbouwh** [reply]
Updated the proposal

**Comment by MarkGodwin** [comment]
Reviving this discussion, as I had a similar thought to this....

If a network management system like Unifi or TP-Link Omada manages a set of wi-fi access points in an environment, it's possible to work out roughly where each wi-fi client on the network is located, based on which access point is being used by each device. From my tests, this works pretty well for giving a rough location for a person in my home (e.g. living room, garden, kitchen) even without fast-roaming APs.

As the Device Tracker entity type includes a Location name attribute, and doesn't _require_ a GPS coordinate, it feels like this nearly fits in with the intended usage. But you may not agree, or it may have unexpected side effects... 
I have a [draft PR](https://github.com/home-assistant/core/pull/115601) open that implements this, but it won't be allowed due to the non-standard use of the Device Tracker entity type (among other things).

So, I think we could solve this with a third type of device tracker, but I'm not sure I'd vote for the name Presence Entity.


Honestly, I'm not sure how much milage there is in creating automations based on approximate locations, but it seems like something that people might find useful for things like lighting or heating, or as a second source of truth, e.g. "Occupancy detected in living room AND only my iPhone close to the living room, then activate Karaoke mode". I'm limited by my imagination for automations.

Another possible problem would be the approximate nature of the location updates based on whether a device decided to roam to the nearer AP might confuse users.


The other idea I had for achieving this was using the Device's area of the current access point, and updating the Area of each Device Scanner based on which AP it has been connected to. However, I'm not sure that the Device Scanners even have Areas, if they can be easily changed dynamically, and how useful that would be for any integrations other than dashboards.
